plugins/module_utils/dnac.py

```diff
@@ -331,7 +332,14 @@ def check_task_response_status(self, response, validation_string, data=False):
             return self
 
         task_id = response.get("taskId")
+        start_time = time.time()
         while True:
+            end_time = time.time()
+            if (end_time - start_time) >= self.max_timeout:
+                self.log("Max timeout of {0} sec has reached for the task id '{1}' for the event and unexpected "
+                         "api status so moving out of the loop.".format(self.max_timeout, task_id), "WARNING")
+                break
+
             task_details = self.get_task_details(task_id)
             self.log('Getting task details from task ID {0}: {1}'.format(task_id, task_details), "DEBUG")
```

Comment:
self.log("Retrieved task details for task ID {0}: {1}".format(task_id, task_details), "DEBUG")

Line 361:
    self.log("Progress is '{0}' for task ID '{1}'".format(task_details.get('progress'), task_id), "DEBUG")